// tablePlanchettesJouees/tablePlanchettesJouees.js
/**
 * Created by Phil on 2018-02-18.
 */

var tablePlanchettesJouees = [];

var nbPlanchettes = 0;
var joue = 0;

module.exports = {
    nbPlanchettes: function(nbP){
        nbPlanchettes = nbP;
    },
    nbEquipes: function(nbE){
        joue = parseInt(nbE)/2;
    },
    creeTable: function(){
        var tableIndex = [];
        var i = 0;
        var j = 0;
        for(i = 0; i < nbPlanchettes; i++){
            var lignePlanchette = [];
            for(j = 0; j <= joue; j++){
                lignePlanchette.push(0);
            }
            tablePlanchettesJouees.push(lignePlanchette);
        }
        //tablePlanchettesJouees.push(tableIndex);
        console.log(tablePlanchettesJouees);
    },
    insertIdPlanchette: function(idP, planchette, indexPartie){

        var indexPl = parseInt(planchette)-1;
        console.log(planchette);
        var indexColonneJouee = parseInt(tablePlanchettesJouees[indexPl].length) - 1;
        var indexJoue = parseInt(tablePlanchettesJouees[indexPl][indexColonneJouee]);
        tablePlanchettesJouees[indexPl][indexJoue] = idP;
        indexJoue += 1;
        tablePlanchettesJouees[indexPl][indexColonneJouee] = indexJoue;
    },
    retourneTable: function(indexPartie){
        return tablePlanchettesJouees;
    },
    retourneIdsPlanchette: function(noPlanchette, indexPartie){
        indexPlanchette = parseInt(noPlanchette)-1;
        return tablePlanchettesJouees[indexPlanchette];
    }
};
