// gestionPartie/gestionPartie_copy.js
/**
 * Created by Phil on 2016-10-22.
 */

//var planchette = {noPl: 0, vulnerabilite: 0, contrat: "", declarant: "", entame: "", resultat: 0, score: 0, termine: 0};
//var ronde = {noRonde: 0, eNS: "", eEO: "", nomsNS: [], nomsEO: [], planchettes: [], termine: 0};
//var table = {noTable: 0, rondes: [], noAcces: 0, occupe: 0, termine: 0};
//var partie = {nom: "", nbEquipes : 0, nbPlanchettes : 0, nbRondes: 0, phamtome : 0, fini : 0, tables: []};
//var index = {noAcces: 0, addr: ""};
var gest = {parties : [], tIndex: []};
var mov = require('../mouvement/mouvement');
var lettre = require('../lettre-equipe/lettreEquipe');
var fs = require('fs');


var noPartie = 0;
var nbTables = 0;

var noms = {nomsLigne1 : [], nomsLigne2 : []};
var varVulPPl = [0,1,2,3,1,2,3,0,2,3,0,1,3,0,1,2];

module.exports = {


    chargePartie: function (nomPartie, infoPartieJson) {
        var indexPartie = partieExiste(nomPartie);
        //console.log(gest);
        mov.initMouvement();
        if (indexPartie >= 0) {
            partieExistante(indexPartie);
            //console.log("Existe");
        } else {
            indexPartie = nouvellePartie(nomPartie, infoPartieJson);
            partieExistante(indexPartie);
            //console.log("Existe pas");
        }
        return indexPartie;
        //console.log(gest.parties[0]);
    },

    retourneTables: function(indexPartie){
        //var partieTemp = trouvePartie(nomPartie);
        return gest.parties[indexPartie].tables;
    },

    retourneNoRonde: function(indexPartie, noTable){
        var tableTemp = gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)];
        var nbRondes = parseInt(gest.parties[indexPartie].nbRondes);
        var i = 0;

        if(rondeTermine(tableTemp, gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overrideRonde) == false)
        {
            return gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overrideRonde;
        }else {
            gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overrideRonde = 0;
            //console.log(nbRondes);
            for (i = 0; i < nbRondes; i++) {
                if (parseInt(tableTemp.rondes[i].termine) != 1) {
                    return tableTemp.rondes[i].noRonde;
                }
            }
        }
        return -1;
    },

    overrideNoRonde: function(indexPartie, noTable, overRonde){
        gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overrideRonde = overRonde;
        gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overridePlanchette = 0;
    },

    overrideNoPlanchette: function(indexPartie, noTable, overRonde, overPlanchette){
        gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overrideRonde = overRonde;
        gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overridePlanchette = parseInt(overPlanchette);
    },

    retourneNoPlanchette: function(indexPartie, noTable, noRonde, overRonde, overPlanchette){
        var rondeTemp = gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].rondes[(parseInt(noRonde)-1)];
        var nbPl = parseInt(rondeTemp.planchettes.length);
        var i = 0;
        var vNoPlanchette = -1;

        if(planchetteTermine(gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].rondes[(parseInt(gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overrideRonde)-1)],
            gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overridePlanchette) == false){
            return gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overridePlanchette;
        }else {
            gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overrideRonde = 0;
            gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overridePlanchette = 0;
            //console.log(nbRondes);
            for (i = 0; i < nbPl; i++) {
                if (parseInt(rondeTemp.planchettes[i].termine) != 1) {
                    return rondeTemp.planchettes[i].noPlanchette;
                }
            }
        }


        /*if(parseInt(gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overridePlanchette) != 0){
            console.log("OverridePl : " + gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overridePlanchette);
            vNoPlanchette = gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overridePlanchette;
            gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overridePlanchette = 0;
            gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].overrideRonde = 0;
        }else {
            for (i = 0; i < nbPl; i++) {
                if (parseInt(rondeTemp.planchettes[i].termine) != 1) {
                    return rondeTemp.planchettes[i].noPlanchette;
                }
            }
        }*/
        return vNoPlanchette;
    },

    retourneENS: function(indexPartie, noTable, noRonde){
        return gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].rondes[(parseInt(noRonde)-1)].eNS;
    },

    retourneEEO: function(indexPartie, noTable, noRonde){
        return gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].rondes[(parseInt(noRonde)-1)].eEO;
    },

    retourneNomsNS: function(indexPartie, noTable, noRonde){
        return gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].rondes[(parseInt(noRonde)-1)].nomsNS;
    },

    retourneNomsEO: function(indexPartie, noTable, noRonde){
        return gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].rondes[(parseInt(noRonde)-1)].nomsEO;
    },

    retourneVul: function(indexPartie, noTable, noRonde, noPlanchette){
        //console.log(indexPartie +" "+ noTable+" "+ noRonde+" "+ noPlanchette);
        return gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].rondes[(parseInt(noRonde)-1)].planchettes[retourneIndexPlanchette(indexPartie, noTable, noRonde, noPlanchette)].vulnerabilite;
    },

    tableTermine: function(indexPartie, noTable){
        //rondesTermine(gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)])
        if(parseInt(gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable)-1)].termine) === 1){
            return true;
        }else{
            return false;
        }
    },

    ecrireResultatPlanchette: function(infoTable, requete){
        var indexPlanchette = retourneIndexPlanchette(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde, infoTable.noPlanchette);
        var planchette = gest.parties[parseInt(infoTable.indexPartie)].tables[(parseInt(infoTable.noTable) - 1)].rondes[(parseInt(infoTable.noRonde) - 1)].planchettes[indexPlanchette];
        var ronde = gest.parties[parseInt(infoTable.indexPartie)].tables[(parseInt(infoTable.noTable) - 1)].rondes[(parseInt(infoTable.noRonde) - 1)];
        var signe = determineSigne(requete.faitSigne);
        var lPointage;
        var index = 0;
        //console.log(planchette.noPlanchette);

        planchette.contrat = requete.nbLeves + ' ' + determineCouleur(requete.couleur);
        planchette.contre = requete.contre;
        planchette.declarant = determineDeclarant(requete.par);
        planchette.entame = retourneEntame(requete.entame) + ' de ' + determineCouleur(requete.entameCouleur);

        if(signe != '='){
            planchette.resultat = signe+requete.fait;
        }else{
            planchette.resultat = signe;
        }

        planchette.score = parseInt(requete.pNS) - parseInt(requete.pEO);
        planchette.termine = 1;

        //console.log(lettre.retourneIndex(ronde.eEO));
        lPointage = retourneLignePointage(planchette.noPlanchette, gest.parties[parseInt(infoTable.indexPartie)].feuillePointage.lignePointage);

        index = lettre.retourneIndex(ronde.eNS);
        lPointage.pointageEquipe[index].score = parseInt(planchette.score);
        lPointage.pointageEquipe[index].orientation = "NS";

        index = lettre.retourneIndex(ronde.eEO);
        lPointage.pointageEquipe[index].score = -parseInt(planchette.score);
        lPointage.pointageEquipe[index].orientation = "EO";

        lPointage.totalJoue += 1;
        //console.log(gest.parties[parseInt(infoTable.indexPartie)].nbEquipes/2);

        if(lPointage.totalJoue == gest.parties[parseInt(infoTable.indexPartie)].nbEquipes/2){
            calculePointage(lPointage);
        }
        //console.log(lPointage);
        if(planchettesTermine(gest.parties[parseInt(infoTable.indexPartie)].tables[(parseInt(infoTable.noTable) - 1)].rondes[(parseInt(infoTable.noRonde) - 1)]) == true){
            gest.parties[parseInt(infoTable.indexPartie)].tables[(parseInt(infoTable.noTable)-1)].rondes[(parseInt(infoTable.noRonde) - 1)].termine = 1;
        }

        if(rondesTermine(gest.parties[parseInt(infoTable.indexPartie)].tables[(parseInt(infoTable.noTable) - 1)]) == true){
            gest.parties[parseInt(infoTable.indexPartie)].tables[(parseInt(infoTable.noTable)-1)].termine = 1;
        }
        /*var obj = gest.parties[parseInt(infoTable.indexPartie)];
        console.log(obj);
        var objStr = JSON.stringify(gest.parties[parseInt(infoTable.indexPartie)]);
        console.log(objStr);
        console.log(JSON.parse(objStr).tables[0].rondes);*/
        sauvgarderPartie(infoTable.indexPartie);
    },

    retourneNomsEquipe: function(infoTable, indexEquipe){
        //var index = lettre.retourneIndex(equipe);
        return gest.parties[parseInt(infoTable.indexPartie)].feuillePointage.noms[indexEquipe];
    },

    retourneNbEquipes: function(infoTable){
        return gest.parties[parseInt(infoTable.indexPartie)].nbEquipes;
    },

    retourneNbPlanchettes: function(infoTable){
        return gest.parties[parseInt(infoTable.indexPartie)].nbPlanchettes;
    },

    retourneScoreEquipe: function(infoTable, i, j){
        return gest.parties[parseInt(infoTable.indexPartie)].feuillePointage.lignePointage[j].pointageEquipe[i].pointage;
    },

    retourneAccesDirectRondes: function(indexPartie, noTable){
        var rondes = gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable) - 1)].rondes;
        var i = 0;
        var j = 0;
        var accDirect = [];
        for(i = 0; i < rondes.length; i++){
            if(parseInt(rondes[i].termine) != 1){
                accDirect.push(rondes[i].noRonde);
            }
        }
        return accDirect;
    },

    retourneAccesDirectPlanchettes: function(indexPartie, noTable, ronde) {
        var planchettes = gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable) - 1)].rondes[parseInt(ronde) - 1].planchettes;
        var i = 0;
        var j = 0;
        var accDirect = [];
        for (i = 0; i < planchettes.length; i++) {
            if (parseInt(planchettes[i].termine) != 1) {
                accDirect.push(planchettes[i].noPlanchette);
            }
        }
        return accDirect;
    },

    sauvgarderPartie: function(indexPartie){
        var objStr = JSON.stringify(gest.parties[parseInt(indexPartie)]);

        fs.writeFile(gest.parties[parseInt(indexPartie)].nom+'.txt', objStr, function (err) {
            if (err) throw err;
            console.log('Saved!');
        });
    }
}

function sauvgarderPartie(indexPartie){
    var objStr = JSON.stringify(gest.parties[parseInt(indexPartie)]);

    fs.writeFile('./fichiers/parties jouees/'+gest.parties[parseInt(indexPartie)].nom+'.txt', objStr, 'utf8', function (err) {
        if (err) throw err;
        console.log('Fichier sauvgardé!');
    });
}

function retourneLignePointage(pl, lp){
    var err = -1;
    var p = 0;
    for(p = 0; p < lp.length; p++){
        if(lp[p].noPlanchette == pl){
            return lp[p];
        }
    }
    return err;
}

function calculePointage(lPointage){
    var score1 = 0;
    var score2 = 0;
    var p1 = 0;
    var p2 = 0;
    //console.log("Calcule");
    for(p1 = 0; p1 < lPointage.pointageEquipe.length; p1++) {
        for(p2 = 0; p2 < lPointage.pointageEquipe.length; p2++) {
            if((p1 != p2) && (lPointage.pointageEquipe[p1].orientation == lPointage.pointageEquipe[p2].orientation)){
                score1 = lPointage.pointageEquipe[p1].score;
                score2 = lPointage.pointageEquipe[p2].score;
                if (score1 > score2) {
                    lPointage.pointageEquipe[p1].pointage = lPointage.pointageEquipe[p1].pointage + 1;
                    /*if ((fantome == true) && (nbTable === 3)) {
                        sNS = sNS + 1;
                    } else if ((fantome == true) && (nbTable === 4)) {
                        sNS = sNS + 0.5;
                    }*/
                } else if (score1 == score2) {
                    lPointage.pointageEquipe[p1].pointage = lPointage.pointageEquipe[p1].pointage + 0.5;

                    /*if ((fantome == true) && (nbTable === 3)) {
                        sNS = sNS + 0.5;
                        sEO = sEO + 0.5;
                    } else if ((fantome == true) && (nbTable === 4)) {
                        sNS = sNS + 0.25;
                        sEO = sEO + 0.25;
                    }*/
                } /*else {
                    sEO = sEO + 1;
                    if ((fantome == true) && (nbTable === 3)) {
                        sEO = sEO + 1;
                    } else if ((fantome == true) && (nbTable === 4)) {
                        sEO = sEO + 0.5;
                    }
                }*/
            }
        }
    }

}

function determineSigne(v){
    switch(v){
        case '2':
            return '+';
        case '3':
            return '-';
        default :
            return '=';
    }

}

function determineCouleur(v){
    switch(v){
        case '1':
            return 'Trèfle';
        case '2':
            return 'Carreau';
        case '3':
            return 'Coeur';
        case '4':
            return 'Pique';
        case '5':
            return 'Sans atout';
        default :
            return 'Autre';
    }

}

function determineDeclarant(v){
    switch(v){
        case '1':
            return 'Nord';
        case '2':
            return 'Sud';
        case '3':
            return 'Est';
        case '4':
            return 'Ouest';
        default :
            return 'Autre';
    }

}

function retourneEntame(entame){
    switch (entame){
        case '1':
            return 'as';
        case '11':
            return 'valet';
        case '12':
            return 'dame';
        case '13':
            return 'roi';
    }
    return entame;
}

function planchettesTermine(ronde){
    var p = 0;
    for(p = 0; p < ronde.planchettes.length; p++){
        if(parseInt(ronde.planchettes[p].termine) != 1){
            return false;
        }
    }
    return true;
}

function rondeTermine(table, noRonde){
    if(parseInt(noRonde) != 0){
        if(parseInt(table.rondes[parseInt(noRonde) - 1].termine) != 1){
            return false;
        }
    }
    return true;
}

function planchetteTermine(ronde, noPlanchette){
    console.log(ronde);
    console.log(noPlanchette);
    if(parseInt(noPlanchette) != 0){
        if(parseInt(ronde.planchettes[parseInt(retourneIndexPlanchetteRonde(ronde, noPlanchette))].termine) != 1){
            return false;
        }
    }
    return true;
}

function rondesTermine(table){
    var r = 0;
    for(r = 0; r < table.rondes.length; r++){
        if(parseInt(table.rondes[r].termine) != 1){
            return false;
        }
    }
    return true;
}

function retourneIndexPlanchetteRonde(ronde, noPlanchette){
    var i = -1;
    var tailleI = ronde.planchettes.length;
    for(i = 0; i < tailleI; i++) {
        if(ronde.planchettes[i].noPlanchette === parseInt(noPlanchette)){
            return i;
        }

    }
    return i;
}

function retourneIndexPlanchette(indexPartie, noTable, noRonde, noPlanchette){
    var i = -1;
    var tailleI = gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable) - 1)].rondes[(parseInt(noRonde) - 1)].planchettes.length;
    for(i = 0; i < tailleI; i++) {
        if(gest.parties[parseInt(indexPartie)].tables[(parseInt(noTable) - 1)].rondes[(parseInt(noRonde) - 1)].planchettes[i].noPlanchette === parseInt(noPlanchette)){
            return i;
        }

    }
    return i;
}

function nouvellePartie(nomPartie, infoPartieJson){
    var indexPartie = 0;
    infoPartieJson.partie.feuille[0].noms[0].nom1[0].nom.shift();
    infoPartieJson.partie.feuille[0].noms[0].nom2[0].nom.shift();
    //console.log(infoPartieJson.partie.feuille[0].noms[0].nom2[0].nom);
    noms.nomsLigne1 = infoPartieJson.partie.feuille[0].noms[0].nom1[0].nom;
    noms.nomsLigne2 = infoPartieJson.partie.feuille[0].noms[0].nom2[0].nom;
    gest.parties.push(new partie(nomPartie,
        infoPartieJson.partie.stats[0].nbEquipes[0],
        infoPartieJson.partie.stats[0].nbPlanchettes[0],
        infoPartieJson.partie.stats[0].nbRondes[0],
        infoPartieJson.partie.stats[0].phantome[0],
        infoPartieJson.partie.stats[0].fini[0],
        [],
        new feuillePointage(creeLignePointage(infoPartieJson.partie.stats[0].nbEquipes[0], infoPartieJson.partie.stats[0].nbPlanchettes[0]), inscritNoms(infoPartieJson.partie.stats[0].nbEquipes[0]))
    ));
    //console.log(gest.parties[0].feuillePointage);
    indexPartie = (parseInt(gest.parties.length)-1);

    creeTables(indexPartie);

    return(indexPartie);
}

function inscritNoms(nbE){
    var tNoms = [];
    var equipe = [];
    var j = 0;
    for(j = 0; j < parseInt(nbE); j++){
        //console.log(noms.nomsLigne1[j]);
        equipe.push(noms.nomsLigne1[j], noms.nomsLigne2[j]);
        tNoms.push(equipe);

        equipe = [];

    }
    //console.log(tNoms);
    return tNoms;
}

function partieExistante(indexPartie){
    var partie = gest.parties[indexPartie];
}

/*function trouvePartie(nomPartie){
    return gest.parties[partieExiste(nomPartie)];
}*/

function partieExiste(nomPartie){
    var vide = -1;
    for(i = 0; i < gest.parties.length; i++){
        if(gest.parties[i].nom === nomPartie){
            return i;
        }
    }
    return vide;
}

function creeLignePointage(nbE, nbP){
    var tLignePointage = [];
    var i = 0;
    for(i = 0; i < parseInt(nbP); i++){
        tLignePointage.push(new lignePointage(i+1, creePointageEquipe(nbE), 0));
    }
    //console.log(tLignePointage[0].pointageEquipe[0]);
    return tLignePointage;
}

function creePointageEquipe(nbE){
    var tPointageEquipe = [];
    var j = 0;
    for(j = 0; j < parseInt(nbE); j++){
        tPointageEquipe.push(new pointageEquipe(lettre.retourneLettre(j+1),0,0,""));
    }
    //console.log(tPointageEquipe);
    return tPointageEquipe;
}

function creeTables(indexPartie){
    var nbTables = 0;
    var indexTable = 0;
    if(gest.parties[indexPartie].phamtome == 1){
        nbTables = ((parseInt(gest.parties[indexPartie].nbEquipes)/2)-1);
    }else{
        nbTables = (parseInt(gest.parties[indexPartie].nbEquipes)/2);
    }

    for(indexTable = 0; indexTable < nbTables; indexTable++){
        gest.parties[indexPartie].tables.push(new table(indexTable+1,[],0,0,0,0,0));
        creeRondes(indexTable, indexPartie);
        //console.log(gest.parties[indexPartie].tables[indexTable].rondes);

    }
}

function creeRondes(indexTable, indexPartie){
    var nbRondes = parseInt(gest.parties[indexPartie].nbRondes);
    var mouvement = mov.retourneMouvement(gest.parties[indexPartie].nbEquipes, indexTable);
    var indexENS = 0;
    var indexEEO = 0;
    var groupePlanchetes = 0;
    var indexRonde = 0;
    //console.log(mouvement);
    for(indexRonde = 0; indexRonde < nbRondes; indexRonde++){
        //console.log(mouvement[j][0]);
        indexENS = mouvement[indexRonde][0];
        indexEEO = mouvement[indexRonde][1];
        groupePlanchetes = mouvement[indexRonde][2];
        gest.parties[indexPartie].tables[indexTable].rondes.push(
            new ronde(indexRonde+1,
                lettre.retourneLettre(indexENS),
                lettre.retourneLettre(indexEEO),
                [noms.nomsLigne1[indexENS-1], noms.nomsLigne2[indexENS-1]],
                [noms.nomsLigne1[indexEEO-1], noms.nomsLigne2[indexEEO-1]],
                [],
                0
            )
        );
        creePlanchettes(indexRonde, indexTable, indexPartie, groupePlanchetes);

    }
    //console.log(gest.parties[indexPartie].tables[indexTable].rondes);
}

function creePlanchettes(indexRonde, indexTable, indexPartie, groupePlanchetes){
    var nbPlParRondes = (parseInt(gest.parties[indexPartie].nbPlanchettes)/parseInt(gest.parties[indexPartie].nbRondes));
    var unePlanchette = 0;
    var ecart = nbPlParRondes - 1;
    var debut = ((nbPlParRondes * parseInt(groupePlanchetes)) - ecart);
    for(unePlanchette = debut; unePlanchette <= (nbPlParRondes * parseInt(groupePlanchetes)); unePlanchette++){
        gest.parties[indexPartie].tables[indexTable].rondes[indexRonde].planchettes.push(
            new planchette(
                unePlanchette,
                varVulPPl[unePlanchette-1],
                "",
                "",
                "",
                "",
                0,
                0,
                0
            )
        );


    }
    //console.log(gest.parties[indexPartie].tables[indexTable].rondes[indexRonde].planchettes);
}

function index(noA, addr){
    this.noAcces = noA;
    this.addresse = addr;
}

function partie(nomFichier, nbE, nbP, nbR, pham, f, tTables, feuillePointage){
    this.nom = nomFichier;
    this.nbEquipes = nbE;
    this.nbPlanchettes = nbP;
    this.nbRondes = nbR;
    this.phamtome = pham;
    this.fini = f;
    this.tables = tTables;
    this.feuillePointage = feuillePointage;
}

function feuillePointage(tLignePointage, tNoms){
    this.lignePointage = tLignePointage;
    this.noms = tNoms;
}

function lignePointage(noPlanchette, tPointageEquipe, totalJoue){
    this.noPlanchette = noPlanchette;
    this.pointageEquipe = tPointageEquipe;
    this.totalJoue = totalJoue;
}

function pointageEquipe(equipe, score, pointage, orientation){
    this.equipe = equipe;
    this.score = score;
    this.pointage = pointage;
    this.orientation = orientation;
}

function table(noT, tRondes, noA, occ, overrideRonde, overridePlanchette, ter){
    this.noTable = noT;
    this.rondes = tRondes;
    this.noAcces = noA;
    this.occupe = occ;
    this.overrideRonde = overrideRonde;
    this.overridePlanchette = overridePlanchette;
    this.termine = ter
}

function ronde(noR, sEquipeNS, sEquipeEO, tNomsNS, tNomsEO, tPl, ter){
    this.noRonde = noR;
    this.eNS = sEquipeNS;
    this.eEO = sEquipeEO;
    this.nomsNS = tNomsNS;
    this.nomsEO = tNomsEO;
    this.planchettes = tPl;
    this.termine = ter;
}

function planchette(noPl, vul, con, ctr, decl, ent, resul, sco, ter){
    this.noPlanchette = noPl;
    this.vulnerabilite = vul;
    this.contrat = con;
    this.contre = ctr;
    this.declarant = decl;
    this.entame = ent;
    this.resultat = resul;
    this.score = sco;
    this.termine = ter;
}

/*function nouvellePartie(nFichier){
    var nouvelle = true;
    var partie = [];
    for(i = 0; i < gest.length; i++){
        partie = gest[i];
        if(partie[0] == nFichier){
            nouvelle = false
        }
        return nouvelle;
    }
}*/ /*getPartie: function(){
 return partie;
 },

 getNoms: function(){
 return noms;
 },

 setPartie: function(nbE, nbP, nbR, score, p, fini){
 partie.nbEquipes = nbE;
 partie.nbPlanchettes = nbP;
 partie.nbRondes = nbR;
 partie.score = score;
 partie.phamtome = p;
 partie.fini = fini;
 },

 setNoms: function(nL1, nL2){
 noms.nomsLigne1 = nL1;
 noms.nomsLigne2 = nL2;
 },

 gestionnaire: function(){

 },

 getNoPartie: function(){
 return noPartie;
 },

 incNoPartie: function(){
 noPartie = noPartie + 1;
 },

 setNbTables: function(nb){
 nbTables = nb;
 },

 chargePartie: function(nFichier){
 if(nouvellePartie(nFichier) == true){

 }else{

 }
 }*/
