// tableau/tableau.js
/**
 * Created by Phil on 2016-02-04.
 */

var tableau = [];
var rangee = [];
var partie = [];
var lettre = require('../lettre-equipe/lettreEquipe');
var phantome = false;

module.exports = {
    creeTableau: function(){
        tableau = [];
        var d = 0;
        if (phantome == true){
            partie[0] = parseInt(partie[0]) + 1;
        }
        for(i = 0; i <= parseInt(partie[1])+2; i++){
            for(j = 0; j <= parseInt(partie[0]); j++){

                if(((i == 0) || (i == 1)) && (j == 0)){
                    rangee.push('Nom '+ (i+1).toString());
                }else if((i == 2) && ( j > 0)){
                    rangee.push('Équipe ' + lettre.retourneLettre(j-1));
                } else if((i == 2) && ( j == 0)){
                    rangee.push('Équipes');
                }
                else if((i > 2) && ( j == 0)){
                    rangee.push((i-2).toString());
                }
                else{
                    rangee.push('');
                }
            }
            tableau.push(rangee);
            rangee = [];
        }
        if(phantome == true) {
            iniPhantome();
        }
        console.log(tableau);
    },

    creePartie: function(nbE, nbP, score){
        partie = [];
        partie.push(nbE);
        partie.push(nbP);
        partie.push(nombreDeRondes());
        partie.push(score);
        phantome = isOdd(nbE);
        partie.push(phantome);
        partie.push(0);
    },

    retournePartie: function(){
        return partie;
    },

    ecrirePartie: function(p){
        partie = p;
    },

    retourneTableau: function(){
        return tableau;
    },

    ecrireTableau: function(t){
        tableau = t;
    },

    retourneTableauPosition: function(i,j){
        return tableau[i][j];
    },

    ecrireTableauPosition: function(i,j, texte){
        tableau[i][j] = texte;
    },
    retourneTableauLigne: function(i){
        return tableau[i];
    },

    ecrireTableauLigne: function(i, ligne){
        tableau[i] = ligne;
    },

    retourneNombreDEquipe: function(){
        return partie[0];
    },
    retourneNombrePlanchette: function(){
        return partie[1];
    },
    retourneNombreRondes: function(){
        return partie[2];
    },
    retourneTypeScore: function(){
        return partie[3];
    },
    retournePhantome: function(){
        return partie[4];
    },
    retourneFini: function(){
        return partie[5];
    },
    retourneJSONTableau: function(){
        return JSON.stringify(tableau);
    }
}

function nombreDeRondes(){
    var nbE = parseInt(partie[0]);
    var nbP = parseInt(partie[1]);

    if(isOdd(nbE)){
        return (nbE);
    }else{
        return ((nbE-1));
    }
}

function iniPhantome() {
    var nbE = parseInt(partie[0]);
    var nbP = parseInt(partie[1]);
    var nbR = parseInt(partie[2]);
    var nbPpR = nbP / nbR;
    var j = 0;
    //console.log(nbR);
    tableau[0][nbE] = 'Phantome';
    tableau[1][nbE] = 'Phantome';
    for(i = 0; i < parseInt(partie[1]); i ++){
        tableau[i+3][nbE] = '1';
        if((i % nbPpR) == 0){
            j++;
        }
        tableau[i+3][j] = '1';
    }
}
function isOdd(num) { return num % 2;}
