// table/table.js
/**
 * Created by Phil on 2018-02-09.
 */

var gestion = require('../gestionPartie/gestionPartie');

var infoTable = {
    indexPartie : -1,
    noTable : 0,
    noRonde : 0,
    noPlanchette : 0,
    eNS : "",
    eEO : "",
    nomsNS : [],
    nomsEO : [],
    vul : 0,
    idPl : ""
};

module.exports = {
    infoTable: function(){
        return infoTable;
    },

    chargeInfoTable: function(req, affiche){
        chargeTable(req, affiche);

    },

    chargeInfoTablePlID: function(req){
        chargeTablePlID(req);

    },

    render: function(res){
        /*res.render('table_new', {
            title: 'Table ' + infoTable.noTable,
            title2: 'Table ' + '<span class="spanEq">' + infoTable.noTable + '</span>',
            table: infoTable.noTable,
            indexPartie: infoTable.indexPartie,
            ronde: infoTable.noRonde,
            planchette: infoTable.noPlanchette,
            equipeNS: '<span id="sNS" class="' + determineVulEquipe(1) + '">' + infoTable.eNS + '</span>',
            equipeEO: '<span id="sEO" class="' + determineVulEquipe(2) + '">' + infoTable.eEO + '</span>',
            nomsNS: '<span class="spanEq">' + infoTable.nomsNS[0] + '</span>et <span class="spanEq">' + infoTable.nomsNS[1] + '</span>',
            nomsEO: '<span class="spanEq">' + infoTable.nomsEO[0] + '</span>et <span class="spanEq">' + infoTable.nomsEO[1] + '</span>',
            accesDirect: accesDirect(),
            rondesEtPlanchettes: rondesEtPlanchettes(),
            idPl: infoTable.idPl
        });*/
        res.render('table_new', {
            title: 'Table ' + infoTable.noTable,
            divMainAjax: divMainAjax()
        });
    },

    inscritsPoints: function(requete){

        //chargeInfoTablePlID(requete, false);
        gestion.ecrireResultatPlanchette(requete);
    },

    divMainAjax: function(){
        return divMainAjax();
    },

    rondesEtPlanchettes: function(){
        return rondesEtPlanchettes();
    },

    indexPartieHtml: function(indexPartie){
        console.log("indexPartieInput");
        return '<input type="hidden" name="indexPartie" value="'+indexPartie+'"/>';
    }
};

function accesDirect(){

 var accDirectRondes = gestion.retourneAccesDirectRondes(infoTable.indexPartie, infoTable.noTable);
 var i = 0;
 var sHTML = '<div class="styled-select1 blue semi-square">'+
 '<select id="selectRonde" name="rondeAD">' +
 '<option value="0" disabled selected>Ronde</option>';
 for(i = 0; i < accDirectRondes.length; i++){
 sHTML += '<option value="'+ accDirectRondes[i] +'">'+ accDirectRondes[i] +'</option>';
 }
 sHTML += '</select></div>'+
 '<div id="dSelectPl" class="styled-select1 grey semi-square">'+
 '<select id="selectPlanchette" name="planchetteAD" disabled>' +
 '<option value="0" disabled selected>Planchette</option>' +
 '</select></div>';
 return sHTML;

 }

function chargeTable(req, affiche){
    console.log("/&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&/");
    console.log("/********** Affiche " + affiche + " ************/");

    infoTable.indexPartie = parseInt(req.body.indexPartie);
    infoTable.noTable = parseInt(req.body.table);
    infoTable.noRonde = gestion.retourneNoRonde(infoTable.indexPartie, infoTable.noTable);
    if(parseInt(infoTable.noRonde) != -1) {
        infoTable.noPlanchette = gestion.retourneNoPlanchette(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde, affiche);
        infoTable.eNS = gestion.retourneENS(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde);
        infoTable.eEO = gestion.retourneEEO(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde);
        infoTable.nomsNS = gestion.retourneNomsNS(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde);
        infoTable.nomsEO = gestion.retourneNomsEO(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde);
        infoTable.vul = gestion.retourneVul(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde, infoTable.noPlanchette);
        infoTable.idPl = gestion.retourneIDPl(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde, infoTable.noPlanchette);
    }else{
        console.log("Else charge");
    }
    /*console.log("/&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&/");
    console.log(infoTable);
    console.log("/&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&/");*/
}

function chargeTablePlID(req){
    var infoPlanchette = gestion.retourneInfoPlanchette(req.body.idPlanchette);
    var indexPartie = infoPlanchette[0];
    var indexTable = infoPlanchette[1];
    var indexRonde = infoPlanchette[2];
    var indexPlanchette = infoPlanchette[3];

    infoTable.indexPartie = parseInt(indexPartie);
    infoTable.noTable = (parseInt(indexTable) + 1);
    infoTable.noRonde = (parseInt(indexRonde) + 1);
    infoTable.noPlanchette = gestion.retourneNoPlanchetteIndex(indexPartie, indexTable, indexRonde, indexPlanchette);
    infoTable.eNS = gestion.retourneENS(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde);
    infoTable.eEO = gestion.retourneEEO(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde);
    infoTable.nomsNS = gestion.retourneNomsNS(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde);
    infoTable.nomsEO = gestion.retourneNomsEO(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde);
    infoTable.vul = gestion.retourneVul(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde, infoTable.noPlanchette);
    infoTable.idPl = gestion.retourneIDPl(infoTable.indexPartie, infoTable.noTable, infoTable.noRonde, infoTable.noPlanchette);

}

function rondesEtPlanchettes(){
    var rondes = gestion.retourneRondes(infoTable.indexPartie, infoTable.noTable);
    //console.log(rondes);
    var i = 0;
    var j = 0;
    var sHTML = "";
    var classes = "";
    var style = "";
    var nbPlanchettes = 0;

    for(i = 0; i < rondes.length; i++){
        //console.log(i);
        nbPlanchettes = rondes[i].planchettes.length;
        //nbPlanchettes++;
        if(rondes[i].noRonde == infoTable.noRonde){
            classes = "divAccesRondes divAccesRondeActive divAccesRondesSel";
            style = 'style="height: '+((nbPlanchettes*41)+35+((nbPlanchettes-1)*3))+'px;"';
        }else if(rondes[i].termine == 1){
            classes = "divAccesRondes divAccesRondeFinie";
            style = 'style="height: 28px;"';
        }else{
            classes = "divAccesRondes";
            style = 'style="height: 28px;"';
        }

        sHTML += '<div id="divAccRonde'+rondes[i].noRonde+'" class="'+classes+'" '+style+' onclick="expandRonde(this)">'+
            '<div>Ronde '+rondes[i].noRonde+'</div>';
        for(j = 0; j < nbPlanchettes; j++){
            //console.log(j);
            if((rondes[i].planchettes[j].noPlanchette == infoTable.noPlanchette)&&(rondes[i].noRonde == infoTable.noRonde)){
                classes = "divAccPlan divAccesPlanActive";
            }else if(rondes[i].planchettes[j].termine == 1){
                classes = "divAccPlan divAccesPlanFinie";
            }else{
                classes = "divAccPlan";
            }
            sHTML += '<form id="form'+ rondes[i].noRonde+rondes[i].planchettes[j].noPlanchette +'"><input name ="idPlanchette" type="hidden" value="'+rondes[i].planchettes[j].idPlanchette+'"><input name ="accesDirect" type="hidden" value="0"><div class="'+classes+'" data-form-id="form'+ rondes[i].noRonde+rondes[i].planchettes[j].noPlanchette +'">Planchette '+rondes[i].planchettes[j].noPlanchette+'</div></form>';

//            sHTML += '<form action="/table" method="post"><input name ="idPlanchette" type="hidden" value="'+rondes[i].planchettes[j].idPlanchette+'"><input name ="accesDirect" type="hidden" value="0"><div class="'+classes+'"><input type="submit" value="Planchette '+rondes[i].planchettes[j].noPlanchette+'" class="inputPlanchettes"></div></form>';
            //sHTML += '<div class="divAccPlan" onclick="accesPlanchette()">Planchette '+j+'</div>';  action="/accesDPl" method="post"
        }
        sHTML += '</div>';
    }

    return sHTML;
}

function determineVulEquipe(eq){
    if(parseInt(infoTable.vul) === 3){
        return "spanEqVul";
    }
    if((parseInt(infoTable.vul) === 2) && (eq === 2)){
        return "spanEqVul";
    }
    if((parseInt(infoTable.vul) === 1) && (eq === 1)){
        return "spanEqVul";
    }
    return "spanEq";
}

function divMainAjax(){
    var sHtml =
        '<div id="divAcces">'+rondesEtPlanchettes()+'</div>' +
        '<div id="divVal">' +
            '<div class="divValidation">' +
                '<span>Veuillez valider le r&eacutesultat de la planchette.</span>' +
            '</div>' +
            '<div class="divValidation">' +
                '<h2 class="title">Table&nbsp;<span class="spanEq">' + infoTable.noTable + '</span></h2>' +
                '<div>' +
                    '<h3 class="h2Table">Ronde&nbsp;<span class="spanEq">'+infoTable.noRonde+'</span></h3>' +
                    '<h3 class="h2Table">Planchette&nbsp;<span class="spanEq">'+infoTable.noPlanchette+'</span></h3>' +
                '</div>' +
                '<div class="divEquipes">' +
                    '<div>' +
                        '<span>&Eacute;quipe&nbsp;</span>' +
                        '<span>' +
                            '<span id="sNS" class="' + determineVulEquipe(1) + '">' + infoTable.eNS + '</span>' +
                        '</span>' +
                        '<span>&nbsp;Nord - Sud</span>' +
                    '</div>' +
                    '<div>' +
                        '<span class="spanEq">' + infoTable.nomsNS[0] + '</span>et <span class="spanEq">' + infoTable.nomsNS[1] + '</span>'+
                    '</div>' +
                '</div>' +
                '<div class="divEquipes">' +
                    '<div>' +
                        '<span>&Eacute;quipe&nbsp;</span>' +
                        '<span>' +
                            '<span id="sEO" class="' + determineVulEquipe(2) + '">' + infoTable.eEO + '</span>'+
                        '</span>' +
                        '<span>&nbsp;Est - Ouest</span>' +
                    '</div>' +
                    '<div>' +
                        '<span class="spanEq">' + infoTable.nomsEO[0] + '</span>et <span class="spanEq">' + infoTable.nomsEO[1] + '</span>'+
                    '</div>' +
                '</div>' +
            '</div>' +
            '<div class="divValidation">' +
                '<span>Contrat :&nbsp;</span>' +
                '<span id="contratValidation"></span>' +
            '</div>' +
            '<div class="divValidation">' +
                '<span>Entame :&nbsp;</span>' +
                '<span id="entameValidation"></span>' +
            '</div>' +
            '<div class="divValidation">' +
                '<span>R&eacute;sultat :&nbsp;</span>' +
                '<span id="resultatValidation"></span>' +
            '</div>' +
            '<div class="divValidation">' +
                '<span class="spanEq">Points</span>' +
                '<div class="dCat">' +
                    '<span>' +
                        '<span id="sNS" class="' + determineVulEquipe(1) + '">' + infoTable.eNS + '</span>' +
                    '</span>' +
                    '<span id="spanPNS" class="txtPoints">0</span>' +
                '</div>' +
                '<div class="dCat">' +
                    '<span>' +
                        '<span id="sEO" class="' + determineVulEquipe(2) + '">' + infoTable.eEO + '</span>'+
                    '</span>' +
                    '<span id="spanPEO" class="txtPoints">0</span>' +
                '</div>' +
            '</div>' +
            '<input id="btnReqAnnuler" type="button" value="Annuler" class="btnAjaxAnnuler"/>' +
            '<input id="btnReqAJAX" type="button" value="Valider" class="btnAjax"/>' +
        '</div>' +
        '<div id="divPasse">' +
            '<div class="divValidation">' +
                '<span>Veuillez valider le r&eacutesultat de la planchette.</span>' +
            '</div>' +
            '<div class="divValidation">' +
                '<h2 class="title">Table&nbsp;<span class="spanEq">' + infoTable.noTable + '</span></h2>' +
                '<div>' +
                    '<h3 class="h2Table">Ronde&nbsp;<span class="spanEq">'+infoTable.noRonde+'</span></h3>' +
                    '<h3 class="h2Table">Planchette&nbsp;<span class="spanEq">'+infoTable.noPlanchette+'</span></h3>' +
                '</div>' +
                '<div class="divEquipes">' +
                    '<div>' +
                        '<span>&Eacute;quipe&nbsp;</span>' +
                        '<span>' +
                            '<span id="sNS" class="' + determineVulEquipe(1) + '">' + infoTable.eNS + '</span>' +
                        '</span>' +
                        '<span>&nbsp;Nord - Sud</span>' +
                    '</div>' +
                    '<div>' +
                        '<span class="spanEq">' + infoTable.nomsNS[0] + '</span>et <span class="spanEq">' + infoTable.nomsNS[1] + '</span>'+
                    '</div>' +
                '</div>' +
                '<div class="divEquipes">' +
                    '<div>' +
                        '<span>&Eacute;quipe&nbsp;</span>' +
                        '<span>' +
                            '<span id="sEO" class="' + determineVulEquipe(2) + '">' + infoTable.eEO + '</span>'+
                        '</span>' +
                        '<span>&nbsp;Est - Ouest</span>' +
                    '</div>' +
                    '<div>' +
                        '<span class="spanEq">' + infoTable.nomsEO[0] + '</span>et <span class="spanEq">' + infoTable.nomsEO[1] + '</span>'+
                    '</div>' +
                '</div>' +
            '</div>' +
            '<div class="divValidation">' +
                '<span>Contrat : passe;</span>' +
            '</div>' +
            '<div class="divValidation">' +
                '<span>Entame : passe;</span>' +
            '</div>' +
            '<div class="divValidation">' +
                '<span>R&eacute;sultat : passe;</span>' +
            '</div>' +
            '<div class="divValidation">' +
                '<span class="spanEq">Points</span>' +
                '<div class="dCat">' +
                    '<span>' +
                        '<span id="sNS" class="' + determineVulEquipe(1) + '">' + infoTable.eNS + '</span>' +
                    '</span>' +
                    '<span id="spanPNS" class="txtPoints">0</span>' +
                '</div>' +
                '<div class="dCat">' +
                    '<span>' +
                        '<span id="sEO" class="' + determineVulEquipe(2) + '">' + infoTable.eEO + '</span>'+
                    '</span>' +
                    '<span id="spanPEO" class="txtPoints">0</span>' +
                '</div>' +
            '</div>' +
            '<input id="btnReqAnnuler" type="button" value="Annuler" class="btnAjaxAnnuler"/>' +
            '<input id="btnReqAJAX" type="button" value="Valider" class="btnAjax"/>' +
        '</div>' +
        '<div class="divPage">' +
            '<h1 class="title">Table&nbsp;<span class="spanEq">' + infoTable.noTable + '</span></h1>' +
            '<div>' +
                '<h2 class="h2Table">Ronde&nbsp;<span class="spanEq">'+infoTable.noRonde+'</span></h2>' +
                '<h2 class="h2Table">Planchette&nbsp;<span class="spanEq">'+infoTable.noPlanchette+'</span></h2>' +
            '</div>' +
            '<div>' +
                '<div class="divEquipes">' +
                    '<div>' +
                        '<span>&Eacute;quipe&nbsp;</span>' +
                        '<span>' +
                            '<span id="sNS" class="' + determineVulEquipe(1) + '">' + infoTable.eNS + '</span>' +
                        '</span>' +
                        '<span>&nbsp;Nord - Sud</span>' +
                    '</div>' +
                    '<div>' +
                        '<span class="spanEq">' + infoTable.nomsNS[0] + '</span>et <span class="spanEq">' + infoTable.nomsNS[1] + '</span>'+
                    '</div>' +
                '</div>' +
                '<div class="divEquipes">' +
                    '<div>' +
                        '<span>&Eacute;quipe&nbsp;</span>' +
                        '<span>' +
                            '<span id="sEO" class="' + determineVulEquipe(2) + '">' + infoTable.eEO + '</span>'+
                        '</span>' +
                        '<span>&nbsp;Est - Ouest</span>' +
                    '</div>' +
                    '<div>' +
                        '<span class="spanEq">' + infoTable.nomsEO[0] + '</span>et <span class="spanEq">' + infoTable.nomsEO[1] + '</span>'+
                    '</div>' +
                '</div>' +
            '</div>' +
            '<form id="formPlanchette">' +
                '<input type="hidden" name="table" value="'+infoTable.noTable+'"/>' +
                '<input type="hidden" name="indexPartie" value="'+infoTable.indexPartie+'"/>' +
                '<input type="hidden" name="idPlanchette" value="'+infoTable.idPl+'"/>' +
                '<div class="divMain">' +
                    '<div id="divEnteteContrat" onclick="afficheContrat()" class="divEntete divEnteteSelect">' +
                        '<div class="divSousEntete">Contrat</div>' +
                    '</div>' +
                    '<div id="divSectionContrat" class="divSection divSectionSel">' +
                        '<div class="divSSBordureBleue"></div>' +
                        '<div id="divSousSectionContrat" class="divSousSection divSousSectionSel divSectionMedia">' +
                            '<div class="divPasse">' +
                                '<input id="btnPasse" type="button" value="Passe" class="btnAjax btnPasse"/>' +
                            '</div>' +
                            '<h1 class="h1Section">Contrat</h1>' +
                            '<ul><li><input type="radio" value="1" name="nbLeves" id="radio1" class="radio radioScore"/><label for="radio1" onclick="selLevees(this)" class="label labelScoreUnSel">1</label></li><li><input type="radio" value="2" name="nbLeves" id="radio2" class="radio radioScore"/><label for="radio2" onclick="selLevees(this)" class="label labelScoreUnSel">2</label></li><li><input type="radio" value="3" name="nbLeves" id="radio3" class="radio radioScore"/><label for="radio3" onclick="selLevees(this)" class="label labelScoreUnSel">3</label></li><li><input type="radio" value="4" name="nbLeves" id="radio4" class="radio radioScore"/><label for="radio4" onclick="selLevees(this)" class="label labelScoreUnSel">4</label></li><li><input type="radio" value="5" name="nbLeves" id="radio5" class="radio radioScore"/><label for="radio5" onclick="selLevees(this)" class="label labelScoreUnSel">5</label></li><li><input type="radio" value="6" name="nbLeves" id="radio6" class="radio radioScore"/><label for="radio6" onclick="selLevees(this)" class="label labelScoreUnSel">6</label></li><li><input type="radio" value="7" name="nbLeves" id="radio7" class="radio radioScore"/><label for="radio7" onclick="selLevees(this)" class="label labelScoreUnSel">7</label></li><div style="display:inline-block; width: 20px;"></div><li><input type="radio" value="1" name="contre" id="radioCtr1" onchange="calcule()" checked="checked" class="radio"/><label for="radioCtr1" onclick="selContre(this)" class="label labelContreSel">&nbsp;</label></li><li><input type="radio" value="2" name="contre" id="radioCtr2" onchange="calcule()" class="radio"/><label for="radioCtr2" onclick="selContre(this)" class="label labelContreUnSel">X</label></li><li><input type="radio" value="3" name="contre" id="radioCtr3" onchange="calcule()" class="radio"/><label for="radioCtr3" onclick="selContre(this)" class="label labelContreUnSel">XX</label></li></ul>' +
                            '<hr/>' +
                            '<ul><li><input type="radio" value="1" name="couleur" id="radioTr" onchange="calcule()" class="radio"/><label for="radioTr"><div onclick="selLeveesCoul(this)" class="divCouleur divCouleurConUnSel"><img src="images/Playing card suits icons/1b.png" class="imgScore"/></div></label></li><li><input type="radio" value="2" name="couleur" id="radioCa" onchange="calcule()" class="radio"/><label for="radioCa"><div onclick="selLeveesCoul(this)" class="divCouleur divCouleurConUnSel"><img src="images/Playing card suits icons/2b.png" class="imgScore"/></div></label></li><li><input type="radio" value="3" name="couleur" id="radioCo" onchange="calcule()" class="radio"/><label for="radioCo"><div onclick="selLeveesCoul(this)" class="divCouleur divCouleurConUnSel"><img src="images/Playing card suits icons/3b.png" class="imgScore"/></div></label></li><li><input type="radio" value="4" name="couleur" id="radioPi" onchange="calcule()" class="radio"/><label for="radioPi"><div onclick="selLeveesCoul(this)" class="divCouleur divCouleurConUnSel"><img src="images/Playing card suits icons/4b.png" class="imgScore"/></div></label></li><li><input type="radio" value="5" name="couleur" id="radioSA" onchange="calcule()" class="radio"/><label for="radioSA"><div onclick="selLeveesCoul(this)" class="divCouleur divCouleurConUnSel">SA</div></label></li></ul>' +
                            '<hr/>' +
                            '<ul><li><input type="radio" value="1" name="par" id="radioN" onchange="calcule()" class="radio radioPar"/><label for="radioN" onclick="selPar(this)" class="label labelParUnSel">Nord</label></li><li><input type="radio" value="2" name="par" id="radioS" onchange="calcule()" class="radio radioPar"/><label for="radioS" onclick="selPar(this)" class="label labelParUnSel">Est</label></li><li><input type="radio" value="3" name="par" id="radioE" onchange="calcule()" class="radio radioPar"/><label for="radioE" onclick="selPar(this)" class="label labelParUnSel">Sud</label></li><li><input type="radio" value="4" name="par" id="radioO" onchange="calcule()" class="radio radioPar"/><label for="radioO" onclick="selPar(this)" class="label labelParUnSel">Ouest</label></li></ul>' +
                        '</div>' +
                    '</div>' +
                    '<div id="divEnteteEntame" onclick="afficheEntame()" class="divEntete divEnteteDisSelect">' +
                        '<div class="divSousEntete">Entame</div>' +
                    '</div>' +
                    '<div id="divSectionEntame" class="divSection divSectionUnSel">' +
                        '<div class="divSSBordureBleue"></div>' +
                        '<div id="divSousSectionEntame" class="divSousSection divSousSectionUnSel">' +
                            '<h1 class="h1Section">Entame</h1>' +
                            '<ul><li><input type="radio" value="2" name="entame" id="radioE2" class="radio radioEntame"/><label for="radioE2" onclick="selEntame(this)" class="label labelEntameUnSel">2</label></li><li><input type="radio" value="3" name="entame" id="radioE3" class="radio radioEntame"/><label for="radioE3" onclick="selEntame(this)" class="label labelEntameUnSel">3</label></li><li><input type="radio" value="4" name="entame" id="radioE4" class="radio radioEntame"/><label for="radioE4" onclick="selEntame(this)" class="label labelEntameUnSel">4</label></li><li><input type="radio" value="5" name="entame" id="radioE5" class="radio radioEntame"/><label for="radioE5" onclick="selEntame(this)" class="label labelEntameUnSel">5</label></li><li><input type="radio" value="6" name="entame" id="radioE6" class="radio radioEntame"/><label for="radioE6" onclick="selEntame(this)" class="label labelEntameUnSel">6</label></li><li><input type="radio" value="7" name="entame" id="radioE7" class="radio radioEntame"/><label for="radioE7" onclick="selEntame(this)" class="label labelEntameUnSel">7</label></li><li><input type="radio" value="8" name="entame" id="radioE8" class="radio radioEntame"/><label for="radioE8" onclick="selEntame(this)" class="label labelEntameUnSel">8</label></li><li><input type="radio" value="9" name="entame" id="radioE9" class="radio radioEntame"/><label for="radioE9" onclick="selEntame(this)" class="label labelEntameUnSel">9</label></li><li><input type="radio" value="10" name="entame" id="radioE10" class="radio radioEntame"/><label for="radioE10" onclick="selEntame(this)" class="label labelEntameUnSel">10</label></li><br class="brEntame"/><li><input type="radio" value="11" name="entame" id="radioEV" class="radio radioEntame"/><label for="radioEV" onclick="selEntame(this)" class="label labelEntameUnSel">Valet</label></li><li><input type="radio" value="12" name="entame" id="radioED" class="radio radioEntame"/><label for="radioED" onclick="selEntame(this)" class="label labelEntameUnSel">Dame</label></li><li><input type="radio" value="13" name="entame" id="radioER" class="radio radioEntame"/><label for="radioER" onclick="selEntame(this)" class="label labelEntameUnSel">Roi</label></li><li><input type="radio" value="1" name="entame" id="radioE1" class="radio radioEntame"/><label for="radioE1" onclick="selEntame(this)" class="label labelEntameUnSel">As</label></li></ul>' +
                            '<hr/>' +
                            '<ul><li><input type="radio" value="1" name="entameCouleur" id="radioTrEn" class="radio"/><label for="radioTrEn"><div onclick="selEntameCouleur(this)" class="divCouleur divCouleurEntameUnSel"><img src="images/Playing card suits icons/1b.png" class="imgScore"/></div></label></li><li><input type="radio" value="2" name="entameCouleur" id="radioCaEn" class="radio"/><label for="radioCaEn"><div onclick="selEntameCouleur(this)" class="divCouleur divCouleurEntameUnSel"><img src="images/Playing card suits icons/2b.png" class="imgScore"/></div></label></li><li><input type="radio" value="3" name="entameCouleur" id="radioCoEn" class="radio"/><label for="radioCoEn"><div onclick="selEntameCouleur(this)" class="divCouleur divCouleurEntameUnSel"><img src="images/Playing card suits icons/3b.png" class="imgScore"/></div></label></li><li><input type="radio" value="4" name="entameCouleur" id="radioPiEn" class="radio"/><label for="radioPiEn"><div onclick="selEntameCouleur(this)" class="divCouleur divCouleurEntameUnSel"><img src="images/Playing card suits icons/4b.png" class="imgScore"/></div></label></li></ul>' +
                        '</div>' +
                    '</div>' +
                    '<div id="divEnteteResultat" onclick="afficheResultat()" class="divEntete divEnteteDisSelect">' +
                        '<div class="divSousEntete">R&eacute;sultat</div>' +
                    '</div>' +
                    '<div id="divSectionResultat" class="divSection divSectionUnSel">' +
                        '<div class="divSSBordureBleue"></div>' +
                        '<div id="divSousSectionResultat" class="divSousSection divSousSectionUnSel">' +
                            '<div id="contratResume" class="divContraResume">' +
                                '<div class="divContraResume">Contrat</div>' +
                                '<div id="sousContraResume" class="divContraResume"></div>' +
                            '</div>' +
                            '<h1 class="h1Section">R&eacute;sultat</h1>' +
                            '<span>Nombres de levées réussies</span>' +
                            '<ul><li><div id="divFait0" onclick="selFait(this)" class="divFait"><input type="radio" value="0" name="fait" id="radioFait0" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait0" for="radioFait0" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">0</label></div></li><li><div id="divFait1" onclick="selFait(this)" class="divFait"><input type="radio" value="1" name="fait" id="radioFait1" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait1" for="radioFait1" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">1</label></div></li><li><div id="divFait2" onclick="selFait(this)" class="divFait"><input type="radio" value="2" name="fait" id="radioFait2" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait2" for="radioFait2" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">2</label></div></li><li><div id="divFait3" onclick="selFait(this)" class="divFait"><input type="radio" value="3" name="fait" id="radioFait3" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait3" for="radioFait3" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">3</label></div></li><li><div id="divFait4" onclick="selFait(this)" class="divFait"><input type="radio" value="4" name="fait" id="radioFait4" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait4" for="radioFait4" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">4</label></div></li><br class="brFaitMobile"><li><div id="divFait5" onclick="selFait(this)" class="divFait"><input type="radio" value="5" name="fait" id="radioFait5" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait5" for="radioFait5" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">5</label></div></li><li><div id="divFait6" onclick="selFait(this)" class="divFait"><input type="radio" value="6" name="fait" id="radioFait6" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait6" for="radioFait6" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">6</label></div></li><li><div id="divFait7" onclick="selFait(this)" class="divFait"><input type="radio" value="7" name="fait" id="radioFait7" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait7" for="radioFait7" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">7</label></div></li><li><div id="divFait8" onclick="selFait(this)" class="divFait"><input type="radio" value="8" name="fait" id="radioFait8" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait8" for="radioFait8" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">8</label></div></li><li><div id="divFait9" onclick="selFait(this)" class="divFait"><input type="radio" value="9" name="fait" id="radioFait9" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait9" for="radioFait9" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">9</label></div></li><br/><li><div id="divFait10" onclick="selFait(this)" class="divFait"><input type="radio" value="10" name="fait" id="radioFait10" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait10" for="radioFait10" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">10</label></div></li><li><div id="divFait11" onclick="selFait(this)" class="divFait"><input type="radio" value="11" name="fait" id="radioFait11" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait11" for="radioFait11" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">11</label></div></li><li><div id="divFait12" onclick="selFait(this)" class="divFait"><input type="radio" value="12" name="fait" id="radioFait12" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait12" for="radioFait12" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">12</label></div></li><li><div id="divFait13" onclick="selFait(this)" class="divFait"><input type="radio" value="13" name="fait" id="radioFait13" onchange="calcule()" class="radio radioFait"/><label class="labelFait"></label><label id="labelFait13" for="radioFait13" onclick="event.stopPropagation()" class="label labelFaitUnSel labelFaitDisplay">13</label></div></li></ul>' +
                            '<div class="dCat">' +
                                '<span class="spanEq">Points</span>' +
                                '<div class="dCat">' +
                                    '<span>' +
                                        '<span id="sNS" class="' + determineVulEquipe(1) + '">' + infoTable.eNS + '</span>' +
                                    '</span>' +
                                    '<input id="pNS" type="text" name="pNS" value="0" class="txtPoints"/>' +
                                '</div>' +
                                '<div class="dCat">' +
                                    '<span>' +
                                        '<span id="sEO" class="' + determineVulEquipe(2) + '">' + infoTable.eEO + '</span>'+
                                    '</span>' +
                                    '<input id="pEO" type="text" name="pEO" value="0" class="txtPoints"/>' +
                                '</div>' +
                            '</div>' +
                            '<input id="btnCon" type="button" value="Continuer" class="btnAjax"/>' +
                        '</div>' +
                    '</div>' +
                '</div>' +
                //'<input id="btnReqAJAX" type="button" value="Valider" class="btnAjax"/>' +
            '</form>' +
        '</div>';
    return sHtml;
}
