// planchette/planchette.js
/**
 * Created by Phil on 2016-01-31.
 */

var partie = [];
var table = [];
var planchette = [];

module.exports = {

    ajouterPlanchette: function(t, eNS, eEO, pl){
        planchette = [];

        table = partie[parseInt(t)-1];

        planchette.push(pl);
        planchette.push(eNS);
        planchette.push(eEO);
        planchette.push(0);

        table.push(planchette);

        partie[parseInt(t)-1] = table;

        console.log(partie);
    },

    ajouterBreak: function(t){
        planchette = [];
        table = partie[parseInt(t)-1];
        table.push("FINR");
        partie[parseInt(t)-1] = table;
    },

    ajouteTable: function(t){
        table = [];
        table.push("table "+ t);
        partie.push(table);
        console.log(partie);
    },
    //retournePlanchette

    initPartie: function(){
        partie = [];
    },

    initTable: function(t){
        table = [];
        table.push("table "+ t);
        partie[parseInt(t)-1] = table;
    },
    getPlNb: function(indice){
        return '1';
    },
    getEquipeNS: function(indice){
        return 'A';
    },
    getEquipeEO: function(indice){
        return 'B';
    },
    getVul: function(indice){
        return 3;
    },
    getPlanchette: function(t, pl){
        table = partie[parseInt(t)-1];

        return table[parseInt(pl)];
    },
    getPartie: function(){
        return partie;
    },
    setScore: function(t, pl, s){

        table = partie[parseInt(t)-1];

        table[parseInt(pl)][3] = s;

        partie[parseInt(t)-1] = table;
    }
}
